### PR TITLE
Task 2 - Create MVC Template

### LINKED ISSUE #2: Create MVC template. Scaffold database with EF.

### LINKED ISSUE #5: Create Asp.net MVC solution. Scaffolded db with Entity Framework

### LINKED ISSUE #24: Build basic architecture for the project

### LINKED ISSUE #25: Connect Entity Framework or another ORM

### PR DESCRIPTION
**Pull request by SergiyVengrin**
Closes #2 
Closes #24 
Closes #25

### CODE REVIEW

File: EmployeestWeb.DAL/Data/EmployeestWebDbContext.cs [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+        }
+
+        public EmployeestWebDbContext(DbContextOptions<EmployeestWebDbContext> options)
+            : base(options)
+        {
+        }
+
+        public virtual DbSet<Event> Events { get; set; } = null!;
+
+        public virtual DbSet<EventMember> EventMembers { get; set; } = null!;
```
Comment (DarkPhoeniix) [inline EmployeestWeb.DAL/Data/EmployeestWebDbContext.cs]: As we discussed earlier - we removed events from out DB. Update the code according to the last version of DB.

File: EmployeestWeb.DAL/Data/EmployeestWebDbContext.cs [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+                entity.Property(e => e.EndTime).HasColumnName("end_time");
+
+                entity.Property(e => e.Name)
+                    .HasMaxLength(50)
+                    .HasColumnName("name");
+
+                entity.Property(e => e.StartTime).HasColumnName("start_time");
+            });
+
+            modelBuilder.Entity<EventMember>(entity =>
```
Comment (DarkPhoeniix) [inline EmployeestWeb.DAL/Data/EmployeestWebDbContext.cs]: Same as above - remove entities we are not going to use

File: EmployeestWeb.DAL/Data/EmployeestWebDbContext.cs [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+
+            modelBuilder.Entity<Task>(entity =>
+            {
+                entity.ToTable("tasks");
+
+                entity.Property(e => e.Id).HasColumnName("id");
+
+                entity.Property(e => e.Description).HasColumnName("description");
+
+                entity.Property(e => e.Estimation).HasColumnName("estimation");
```
Comment (DarkPhoeniix) [inline EmployeestWeb.DAL/Data/EmployeestWebDbContext.cs]: We already have story points for the estimation - remove the 'estimation' field

File: EmployeestWeb/Properties/launchSettings.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
+﻿{
+  "iisSettings": {
+    "windowsAuthentication": false,
+    "anonymousAuthentication": true,
+    "iisExpress": {
+      "applicationUrl": "http://localhost:14450",
+      "sslPort": 44391
```
Comment (DarkPhoeniix) [inline EmployeestWeb/Properties/launchSettings.json]: Can we remove the ssl?

**Review by DarkPhoeniix — CHANGES_REQUESTED** [review]

File: EmployeestWeb/Program.cs [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                    npgsqlOptions.EnableRetryOnFailure(
+                        maxRetryCount: 3,
+                        maxRetryDelay: TimeSpan.FromSeconds(5),
+                        errorCodesToAdd: new List<string> { "4060" });
+                }));
+
+            builder.Services.AddControllersWithViews();
+
+
+            builder.Services.AddControllersWithViews();
```
Comment (DarkPhoeniix) [inline EmployeestWeb/Program.cs]: remove the duplicate line

File: EmployeestWeb/Program.cs
Comment (DarkPhoeniix) [inline EmployeestWeb/Program.cs:1]: fix Sonar code smells

File: EmployeestWeb.DAL/Data/EmployeestWebDbContext.cs [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+        public EmployeestWebDbContext()
+        {
+        }
+
+        public EmployeestWebDbContext(DbContextOptions<EmployeestWebDbContext> options)
+            : base(options)
+        {
+        }
+
+        public virtual DbSet<Permission> Permissions { get; set; } = null!;
```
Comment (DarkPhoeniix) [inline EmployeestWeb.DAL/Data/EmployeestWebDbContext.cs]: Use 'default' instead of 'null!'

**Review by DarkPhoeniix — CHANGES_REQUESTED** [review]

File: EmployeestWeb/Program.cs [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+                {
+                    npgsqlOptions.EnableRetryOnFailure(
+                        maxRetryCount: 3,
+                        maxRetryDelay: TimeSpan.FromSeconds(5),
+                        errorCodesToAdd: new List<string> { "4060" });
+                }));
+
+            builder.Services.AddControllersWithViews();
+
+            builder.Services.AddDbContext<EmployeestWebDbContext>(options =>
```
Comment (DarkPhoeniix) [inline EmployeestWeb/Program.cs]: Does it duplicate lines 22-29? If yes, could you remove it?

**Review by DarkPhoeniix — CHANGES_REQUESTED** [review]

**Review by DarkPhoeniix — APPROVED** [review]

**Review by OleksiiDatsiuk — APPROVED** [review]

**Pull request merged**